sw/device/rom_exts/rom_ext_manifest_parser.c

```diff
@@ -0,0 +1,117 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// TODO - parser accesses should be done through mmio and memory libraries.
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/rom_exts/manifest.h"
+
+uint32_t rom_ext_identifier_get(mmio_region_t base_addr) {
+  return mmio_region_read32(base_addr, ROM_EXT_MANIFEST_IDENTIFIER_OFFSET);
+}
+
+bool rom_ext_signature_get(mmio_region_t base_addr, rom_ext_signature_t *dst) {
+  if (dst == NULL) {
+    return false;
+  }
```

Comment:
I feel like this is a slightly silly reason to be returning an error. I feel that the cost of returning by-value here is small (if this was C++, you would be able to count on NRVO, but, alas).

Reply:
`dst` is a pointer to a buffer, which is why we're not returning by value, and it makes sense to ensure it's nonnull before the memcpy.

Reply:
The callers pass the address to where buffer should be copied, hence pass by pointer.

Reply:
The reasoning behind is that it makes the lifetime of large buffers short lived, as the caller would typically define this structure on stack, which will then be clobbered when the calling function returns.